src/app/CommandHandler.cpp

```diff
@@ -64,90 +64,87 @@ CHIP_ERROR CommandHandler::AllocateBuffer()
-
-    if (mTimedRequest != isTimedInvoke)
-    {
-        // The message thinks it should be part of a timed interaction but it's
-        // not, or vice versa.  Spec says to Respond with UNSUPPORTED_ACCESS.
-        err = StatusResponse::Send(Protocols::InteractionModel::Status::UnsupportedAccess, mpExchangeCtx,
-                                   /* aExpectResponse = */ false);
-
-        if (err != CHIP_NO_ERROR)
-        {
-            // We have to manually close the exchange, because we called
-            // WillSendMessage already.
-            mpExchangeCtx->Close();
-        }
-
-        // Null out the (now-closed) exchange, so that when we try to
-        // SendCommandResponse() later (when our holdoff count drops to 0) it
-        // just fails and we don't double-respond.
-        mpExchangeCtx = nullptr;
-        return err;
-    }
 
+    VerifyOrReturnError(invokeRequestMessage.GetSuppressResponse(&mSuppressResponse) == CHIP_NO_ERROR,
+                        Protocols::InteractionModel::Status::InvalidAction);
+    VerifyOrReturnError(invokeRequestMessage.GetTimedRequest(&mTimedRequest) == CHIP_NO_ERROR,
+                        Protocols::InteractionModel::Status::InvalidAction);
+    VerifyOrReturnError(invokeRequestMessage.GetInvokeRequests(&invokeRequests) == CHIP_NO_ERROR,
+                        Protocols::InteractionModel::Status::InvalidAction);
+    VerifyOrReturnError(mTimedRequest == isTimedInvoke, Protocols::InteractionModel::Status::UnsupportedAccess);
     invokeRequests.GetReader(&invokeRequestsReader);
 
     {
         // We don't support handling multiple commands but the protocol is ready to support it in the future, reject all of them and
         // IM Engine will send a status response.
         size_t commandCount = 0;
         TLV::Utilities::Count(invokeRequestsReader, commandCount, false /* recurse */);
-        VerifyOrReturnError(commandCount == 1, CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+        VerifyOrReturnError(commandCount == 1, Protocols::InteractionModel::Status::Failure);
     }
 
     while (CHIP_NO_ERROR == (err = invokeRequestsReader.Next()))
     {
-        VerifyOrReturnError(TLV::AnonymousTag() == invokeRequestsReader.GetTag(), CHIP_ERROR_INVALID_TLV_TAG);
+        VerifyOrReturnError(TLV::AnonymousTag() == invokeRequestsReader.GetTag(),
+                            Protocols::InteractionModel::Status::InvalidAction);
         CommandDataIB::Parser commandData;
-        ReturnErrorOnFailure(commandData.Init(invokeRequestsReader));
+        VerifyOrReturnError(commandData.Init(invokeRequestsReader) == CHIP_NO_ERROR,
+                            Protocols::InteractionModel::Status::InvalidAction);
 
         if (mpExchangeCtx->IsGroupExchangeContext())
         {
-            ReturnErrorOnFailure(ProcessGroupCommandDataIB(commandData));
+            VerifyOrReturnError(ProcessGroupCommandDataIB(commandData) == CHIP_NO_ERROR,
```

Comment:
Will this trigger a response from the device ? If so that's against the specs for Group Communication.

Reply:
no, it would not rigger, when it fails, the error would be returned back to InteractionModelEngine's onMessageReceived, where we would send message when it is not group exchange, thanks
    if (status != Status::Success && !apExchangeContext->IsGroupExchangeContext())
    {
        return StatusResponse::Send(status, apExchangeContext, false /*aExpectResponse*/);
    }